lib/active_job_status/hooks.rb

```diff
@@ -2,12 +2,54 @@ module ActiveJobStatus
   module Hooks
     def self.included(base)
       base.class_eval do
-        before_enqueue { job_tracker.enqueued }
 
-        before_perform { job_tracker.performing }
+        around_enqueue do |job, block|
+          begin
+            before_enqueue(job)
+            job_tracker.enqueued
+            block.call
+            on_enqueue_success(job)
+          rescue Exception => exception
```

Comment:
I'm a bit concerned that we might be swallowing useful information here. Perhaps I'm mistaken, but this seems to catch all exceptions, pass them off to a block (if supplied), and then re-raises a generic exception. At a minimum we'd need to re-raise the exception that was actually thrown, so this doesn't swallow them.